cpp_exercise/tests/linuxipc_test.cc

```diff
@@ -0,0 +1,183 @@
+#include "src/lib/linuxipclib.h"
+#include "src/lib/filehandlerlib.h"
+#include "src/lib/ipcpipelib.h"
+
+#include <map>
+#include <vector>
+#include<thread>
+
+#include "gtest/gtest.h"
+
+
+
+class FileHandlerTests : public ::testing::Test{
+
+      public:
+
+      protected:
+      FileHandler nonexFile{"thisfiledoesnotexist.txt"};
+      FileHandler txtFile{"testfile.txt"};
+      FileHandler pdfFile{"pdftestfile.pdf"};
+      std::string samplestring = "Test Data";
+      std::vector<unsigned char> samplevec;
+      
+    
+ };
+
+
+
+  TEST_F(FileHandlerTests, ReadingFromNoExistingFile){
+
+    remove("thisfiledoesnotexist.txt");
+    std::string exception;
+
+     try{
+       nonexFile.readFile(10);
+       
+       }catch(IPCException & e){
+          exception = e.what();
+          ASSERT_EQ("FileHandler ERROR: Cannot Close the File Correctly", exception);
+          
+       }
+
+    
+  }
+
+   TEST_F(FileHandlerTests, GettingTheCorrectSize){
+     samplevec.insert(samplevec.begin(), samplestring.begin(), samplestring.end());
+    try{
+        txtFile.createFile();
+        txtFile.writeFile(samplevec,samplevec.size());
+    } catch(IPCException & e){}
```

Comment:
I guess it shouldn't throw is size is correct? Similar issue on test below

Reply:
Sorry I did not get this comment completely

Reply:
The test name is 'GettingTheCorrectSize' and you check if sizes of txtFiles and samplevec are equal, so from where exception comes from?

Reply:
Oh u mean that there is no need for try and catch in these two test cases